### ISSUE TITLE
Parse error shows wrong source lines

### ISSUE THREAD

**Issue by crockeea** [ndmitchell/hlint#245]
In the example below, hlint (HEAD) reports:

```
TensorTests.hs:324:22: Error: Parse error: ,
Found:
                 RebindableSyntax, ScopedTypeVariables, TypeFamilies, TypeOperators,
                 UndecidableInstances #-}
  > 
    module TensorTests (tensorTests) where
  

1 hint
```

The line number reported (342) is correct, as is the character with the "parse error" (essentially, #227). The problem is that the source lines are wrong: they are from the top of the file. I traced the problem to `parseModuleEx` in `src/HSE/All.hs`. 

The relevant snippet is
```
ParseFailed sl msg -> do
  -- figure out the best line number to grab context from, by reparsing
  flags <- return $ parseFlagsNoLocations flags -- QUESTIONABLE
  ppstr2 <- runCpp (cppFlags flags) file str
  let pe = case parseFileContentsWithMode (mode flags) ppstr2 of
                   ParseFailed sl2 _ -> context (srcLine sl2) ppstr2
                    _ -> context (srcLine sl) ppstr
  return $ Left $ ParseError sl msg pe
```
There, you reparse the file on a `ParseFailed` with different flags. `sl` is correct (`SrcLoc "TensorTests.hs" 324 22`), but `sl2` is incorrect (`SrcLoc "TensorTests.hs" 1 32`). I'm not sure what the reason is for changing the flags, but that seems to be the problem.

Rename to TensorTests.hs: [TensorTests.txt](https://github.com/ndmitchell/hlint/files/303737/TensorTests.txt)

**Comment by ndmitchell** [comment]
The reparsing with different flags is deliberate. First time around I want the LINE pragmas inserted so I get the correct line numbers. Second time around I don't want that, since I want to grab context from the actual Haskell file, not wherever it is pretending to come from.

However, as part of #154 and #161 I stripped the leading LINE pragma. If there wasn't a leading line pragma, I then stripped the module name, and it failed to parse - as you see above. Now fixed, and I'll make a fresh release, probably tomorrow. Thanks for spotting and reporting it!
